discovery/stackit/stackit.go

```diff
@@ -179,5 +186,10 @@ func newRefresher(conf *SDConfig, l *slog.Logger) (refresher, error) {
 	if conf.Role == RoleServer {
 		return newServerDiscovery(conf, l)
 	}
+
+	if conf.Role == RolePostgresFlex {
+		return newPostgresFlexDiscovery(conf, l)
```

Comment:
If we have more than one role, switch/case should be consider. Like here: https://github.com/jkroepke/prometheus/blob/2aaafae36fc0ba53b3a56643f6d6784c3d67002a/discovery/hetzner/hetzner.go#L160-L174